rocketpool/node/collectors/node-collector.go

```diff
@@ -339,6 +352,33 @@ func (collector *NodeCollector) Collect(channel chan<- prometheus.Metric) {
 		return nil
 	})
 
+	// get the beacon client sync status:
+	wg.Go(func() error {
+		progress := float64(0)
+		
+		syncStatus, err := collector.bc.GetSyncStatus()
+
+		if err != nil {
+			// don't return an error because then the metric isn't emitted at all, just let the Progress be 0
+			fmt.Printf("error getting beacon chain sync status: %w", err)
+		} else {
+			progress = syncStatus.Progress
+		}
+
+		channel <- prometheus.MustNewConstMetric(
```

Comment:
FYI: Note that the channel is used here asynchronously rather than at the bottom of the outer function because this function aborts when the client's aren't synced (since most of the other calculations here require a synced client). I did some testing and this allows these metrics to be emitted even when the client isn't synced while the other ones are not emitted at all on the scrape endpoint until the clients are synced.

Reply:
Yep, noted in: https://github.com/rocket-pool/smartnode/issues/186

Just hasn't been a priority to make a better dependency graph here.

Add comments to the code?

Reply:
Ha. I didn't see that issue. FYI Async publishing seemed to work just fine here. I tested by doing things like shutting down the BC and scrape /metrics the endpoint with wget rapidly and all seemed to behave fine.